packages/functions/src/schema/index.ts

```diff
@@ -56,7 +58,9 @@ export default createRulesetFunction<unknown, Options>(
 
     try {
       let validator;
-      const dialect = opts?.dialect ?? detectDialect(schemaObj) ?? 'draft7';
+      const dialect =
+        (opts.dialect === void 0 || opts.dialect === 'auto' ? detectDialect(schemaObj) : opts?.dialect) ?? 'draft7';
```

Comment:
If `opts` is not defined, it will throw on L56

![image](https://user-images.githubusercontent.com/7423098/129642446-700015c4-f0a5-4830-b346-f5ca5a504fe3.png)


```suggestion
        (opts.dialect === void 0 || opts.dialect === 'auto' ? detectDialect(schemaObj) : opts.dialect) ?? 'draft7';
```